### ISSUE TITLE
meson introspect --targets has no linker information

### ISSUE THREAD

**Issue by nornagon** [mesonbuild/meson#5333]
I'm inspecting a third-party project that's using meson and I'd like to be able to see what all is going to be built and how. `meson introspect --targets` is super useful, but it only contains information about the compilation step, and has no information about the link step and what is linked to what. Is there some other option that I'm missing to generate this information? I don't see anything that looks like the link information in `meson introspect -a` even :\

**Comment by mensinda** [comment]
No, providing linker information is (currently) not implemented in the meson introspection API. This API is primarily meant to provide an IDE with the information required to do code completion, etc. Being able to build a meson project with this information is *not* the goal.

That being said, all compiled source files of a target should be linked together (at least for C, C++, Vala). If you need more information than you have to look at the `meson.build` file(s) of that project or read the generated `ninja.build`.

**Comment by null77** [comment]
How hard would it be to export linker information in Meson introspection? This would be a huge help for integrating Meson into build systems that aren't as flexible. With my integration in Chromium infrastructure, due to all the platforms, it's hard for me to run meson directly and work with our infra's limitations. I would like to use introspection to determine some source information, but currently I can't get what static libraries build into other shared libraries.

@xavierclaessens FYI

**Comment by bonzini** [comment]
Fixed by #11627
